ui/app/utils/path-tree.js

```diff
@@ -0,0 +1,123 @@
+// @ts-check
```

Comment:
The create-a-tree-from-paths-array class. 

Introduces a few JSDoc annotations to the codebase that I hope we can expand as well.